card/mod/05_standard/set/type/search_type.rb

```diff
@@ -38,7 +38,7 @@ def get_query params={}
   if default_limit = query.delete(:default_limit) and !query[:limit]
     query[:limit] = default_limit
   end
-  query[:context] ||= (cardname.junction? ? cardname.left_name : cardname)
+  query[:context] ||= cardname
```

Comment:
If I understand, this gets rid of a great ugliness.  With the typical *right rule virtual card, the context card used to be left, not self.  I guess also when I create a card A+my_tag -> a Search card.  Seems like it may have been a bit broken on *self rules.
The migration looks a little scary, hope it works for all the cases ;)